asserts/account_key_test.go

```diff
@@ -419,3 +421,61 @@ func (aks *accountKeySuite) TestPrerequisites(c *C) {
 		PrimaryKey: []string{"acc-id1"},
 	})
 }
+
+func (aks *accountKeySuite) TestKeyProofHappy(c *C) {
+	kproof, err := asserts.SignWithoutAuthority(asserts.KeyProofType,
+		map[string]interface{}{
+			"nonce": "NONCE",
+			"key":   aks.pubKeyEncoded,
+		}, nil, aks.privKey)
+	c.Assert(err, IsNil)
+
+	// roundtrip
+	a, err := asserts.Decode(asserts.Encode(kproof))
+	c.Assert(err, IsNil)
+
+	kproof2, ok := a.(*asserts.KeyProof)
+	c.Assert(ok, Equals, true)
+
+	// standalone signature check
+	err = asserts.SignatureCheck(kproof2, aks.privKey.PublicKey())
+	c.Check(err, IsNil)
+
+	c.Check(kproof2.Nonce(), Equals, "NONCE")
+}
+
+func (aks *accountKeySuite) TestKeyProofDecodeInvalid(c *C) {
+	encoded := "type: key-proof\n" +
+		"nonce: NONCE\n" +
+		"key:\n    KEY\n" +
+		"body-length: 0\n" +
+		"sign-key-sha3-384: " + aks.privKey.PublicKey().ID() + "\n\n" +
+		"AXNpZw=="
+
+	invalidTests := []struct{ original, invalid, expectedErr string }{
+		{"nonce: NONCE\n", "", `"nonce" header is mandatory`},
+		{"nonce: NONCE\n", "nonce: \n", `"nonce" header should not be empty`},
+		{"key:\n    KEY\n", "", `"key" header is mandatory`},
+		{"key:\n    KEY\n", "key: \n", `"key" header should not be empty`},
+		{"key:\n    KEY\n", "key: $$$\n", `cannot decode public key: .*`},
+	}
+
+	for _, test := range invalidTests {
+		invalid := strings.Replace(encoded, test.original, test.invalid, 1)
+		invalid = strings.Replace(invalid, "KEY", strings.Replace(aks.pubKeyEncoded, "\n", "\n    ", -1), 1)
+		_, err := asserts.Decode([]byte(invalid))
+		c.Check(err, ErrorMatches, keyProofErrPrefix+test.expectedErr)
+	}
+}
+
+func (aks *accountKeySuite) TestKeyProofDecodeKeyIDMismatch(c *C) {
+	invalid := "type: key-proof\n" +
```

Comment:
This assertion does't smell quite right. What are we proving here? The only thing we know is that whoever got that nonce at some point decided to sign an assertion with its private key. It's not binding any information to anything else. Yes, _somebody_ has that private key. :)